pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1489,7 +1490,7 @@ func CheckNodeConditionPredicate(pod *v1.Pod, meta algorithm.PredicateMetadata,
 		}
 	}
 
-	if node.Spec.Unschedulable {
+	if node.Spec.Unschedulable || taints.TaintExistsWithEffect(node.Spec.Taints) {
```

Comment:
I don't think this is correct. We must ensure that the Pod does not tolerate the taint before returning an error and we already do that [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithm/predicates/predicates.go#L1406).

Reply:
@bsalamat - If a user applies a taint without enabling the toleration predicate, we wouldn't respect the taints, can we add it as a mandatory predicate ? cc @gmarek

Reply:
I don't think taint predicate should be a mandatory one. Maybe users want to disable it for reasons which are not clear to us.

Reply:
>  If a user applies a taint without enabling the toleration predicate, we wouldn't respect the taints

It's up to user's configuration; similar to other predicates, e.g. Pod/Node Affinity.

Reply:
BTW, if we want to replace `Unschedulable`, we need to define the taint that we're go to apply to the node.

Reply:
Ok, thank you all for your review, should I close this PR ?